pkgs/applications/networking/cluster/argocd/default.nix

```diff
@@ -1,40 +1,67 @@
-{ lib, buildGoModule, fetchFromGitHub, packr }:
+{ lib, buildGoModule, fetchFromGitHub, packr, makeWrapper, installShellFiles }:
 
 buildGoModule rec {
   pname = "argocd";
-  version = "1.8.6";
-  commit = "28aea3dfdede00443b52cc584814d80e8f896200";
+  version = "2.0.1";
+  commit = "33eaf11e3abd8c761c726e815cbb4b6af7dcb030";
+  tag = "v${version}";
 
   src = fetchFromGitHub {
     owner = "argoproj";
     repo = "argo-cd";
-    rev = "v${version}";
-    sha256 = "sha256-kJ3/1owK5T+FbcvjmK2CO+i/KwmVZRSGzF6fCt8J9E8=";
+    rev = tag;
+    sha256 = "sha256-j/RdiMeaYxlmEvo5CKrGvkp25MrFsSYh+XNYWNcs0PE=";
   };
 
-  vendorSha256 = "sha256-rZ/ox180h9scocheYtMmKkoHY2/jH+I++vYX8R0fdlA=";
+  vendorSha256 = "sha256-8j5v99wOHM/SndJwpmGWiCFEyw4K513HEEbkPrD8C90=";
 
-  doCheck = false;
-
-  nativeBuildInputs = [ packr ];
-
-  buildFlagsArray = ''
-     -ldflags=
-      -X github.com/argoproj/argo-cd/common.version=${version}
-      -X github.com/argoproj/argo-cd/common.buildDate=unknown
-      -X github.com/argoproj/argo-cd/common.gitCommit=${commit}
-      -X github.com/argoproj/argo-cd/common.gitTreeState=clean
-  '';
+  nativeBuildInputs = [ packr makeWrapper installShellFiles ];
 
   # run packr to embed assets
   preBuild = ''
     packr
   '';
 
+  buildFlagsArray =
+    let package_url = "github.com/argoproj/argo-cd/v2/common"; in
+    [
+      "-ldflags="
+      "-s -w"
+      "-X ${package_url}.version=${version}"
+      "-X ${package_url}.buildDate=unknown"
+      "-X ${package_url}.gitCommit=${commit}"
+      "-X ${package_url}.gitTag=${tag}"
+      "-X ${package_url}.gitTreeState=clean"
+    ];
+
+  doCheck = false; # If enabled errors as: https://gist.github.com/superherointj/fe06eeba58ed19b5569cef495b18697c
```

Comment:
Can we try adding helm, kustomize to checkInputs?

Reply:
My attempt to use: `dep2nix -i Gopkg.lock -o deps.nix` failed because one dependency (https://github.com/heptio/validate) is no longer available.

Reply:
Yeah, that seems tricky and to much work. Can we just disable the test(s) that use ksonnet instead of disabling them entirely?

Reply:
I don't know. The [Makefile](https://github.com/argoproj/argo-cd/blob/master/Makefile) has individual tests. I don't understand how Go tests works for such case. How would you do it?

But I'd prefer to not do this for now.

Reply:
I think they are called on by one which is odd. Normally this is done differently in go.